### PR TITLE
[dagster-pipes] start PipesLogReader right away

### PR DESCRIPTION
**Pull request by danielgafni**
## Summary & Motivation

Looks like there is no reason to have `PipesLogReader` startup be dependent on `PipesMessageReader` startup. 

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by danielgafni** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25026?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25026** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25026?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24884** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24884?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#23998](https://github.com/dagster-io/dagster/pull/23998) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23998?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#24469** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24469?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#24809](https://github.com/dagster-io/dagster/pull/24809) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24809?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#25020](https://github.com/dagster-io/dagster/pull/25020) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25020?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#24968** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24968?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24948** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24948?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @danielgafni and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/pipes/utils.py [hunk 1]
```diff
@@ -444,19 +444,9 @@ def _logs_thread(
...
-            # reader threads.
-            elif not messages_thread.is_alive():
-                return
-            time.sleep(DEFAULT_SLEEP_INTERVAL)
-
-        # Logs are started with a merge of the params generated by the message reader and the opened
-        # payload.
-        log_params = {**params, **self.opened_payload}
+        # Logs are started with a merge of the params generated by the message reader, the opened
+        # payload, and params from the state
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/pipes/utils.py:448]: "params from the state" no longer applies

File: python_modules/dagster/dagster/_core/pipes/utils.py [hunk 2]
```diff
@@ -444,19 +444,9 @@ def _logs_thread(
...
+        log_params = {**params, **(self.opened_payload or {})}
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/pipes/utils.py:449]: These are immediately set at the top of the loop so why set them here at all? They aren't used in a call before the loop starts.

File: python_modules/dagster/dagster/_core/pipes/utils.py [hunk 3]
```diff
@@ -497,12 +490,13 @@ def _logs_thread(
...
                     ):
                         for key, log_reader in unstarted_log_readers.items():
-                            warnings.warn(
-                                f"[pipes] Attempted to read log for reader {key}:{log_reader.name} but log was"
-                                f" still not written {WAIT_FOR_LOGS_TIMEOUT} seconds after session close. Abandoning reader {key}."
-                            )
-
-                        return
+                            if key not in failed_log_readers:
+                                failed_log_readers.add(key)
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/pipes/utils.py:494]: Confused about `failed_log_readers` here. Seems like it's not doing anything? If there is only one loop over `unstarted_log_readers` keys, how could we ever loop over the same key twice (and so why would we need to track it in `failed_log_readers`)?

Comment (danielgafni) [reply python_modules/dagster/dagster/_core/pipes/utils.py]: was an old variable leaked from a different PR. Removed it.

File: python_modules/dagster/dagster/_core/pipes/utils.py [hunk 4]
```diff
@@ -466,16 +452,19 @@ def _logs_thread(
             unstarted_log_readers = {**self.log_readers}
 
             while True:
+                if self.opened_payload is not None:
+                    params = {**params, **self.opened_payload}
+
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/pipes/utils.py:457]: Pls modify to only update params once:

```
params_updated = False
while True:
    
     # params are updated once the opened_payload becomes available
     if not params_updated and self.opened_payload is not None:
         params = {**params, **self.opened_payload}
         params_updated = True

...
```

**Review by smackesey — APPROVED** [review]
Pls implement requested change before merge

**Comment by danielgafni** [comment]
### Merge activity

* **Oct 10, 10:41 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/25026) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**